src/snowflake/cli/_plugins/workspace/manager.py

```diff
@@ -45,6 +48,10 @@ def perform_action(self, entity_id: str, action: EntityActions):
         """
         entity = self.get_entity(entity_id)
         if entity.supports(action):
-            getattr(entity, action)()
+            action_ctx = ActionContext(project_root=self.project_root())
+            return getattr(entity, action)(action_ctx)
```

Comment:
if `supports` is a primitive for all entities, shouldn't something like `perform` be defined as well? I'd expect this to look like:
```
action_ctx = ...
if entity.supports(action):
   return entity.perform(action, action_ctx)
```

Reply:
Changed. (This API is probably temporary and will soon be replaced with action decorators)